src/wallet/rpcwallet.cpp

```diff
@@ -4458,7 +4453,7 @@ UniValue z_mergetoaddress(const UniValue& params, bool fHelp)
     CMutableTransaction contextualTx = CreateNewContextualCMutableTransaction(
         Params().GetConsensus(),
         nextBlockHeight);
-    bool isShielded = numNotes > 0 || isToZaddr;
+    bool isShielded = numNotes > 0 || isToSproutZaddr;
```

Comment:
This ends up getting confused in the next commit as it should only be looking at Sprout notes. This is not a problem per se because the `contextualTx` below is only used if there are no sapling notes (currently). I am rebasing and fixing this to make it more clear what is going on.